### PR TITLE
fixing package_id bugs and tests

### PR DESCRIPTION
**Pull request by memsharded**
- Some minor cleanings and simplifications on package_id RequirementInfo
- Fixed one bug on package_id computation for applications

### CODE REVIEW

File: conans/client/graph/compute_pid.py [hunk 1]
```diff
@@ -25,16 +25,18 @@ def compute_package_id(node, new_config):
     for require, transitive in node.transitive_deps.items():
         dep_package_id = require.package_id_mode
```
Comment (lasote) [inline conans/client/graph/compute_pid.py]: Side note, the `dep_package_id` variable is a very confusing name when it is the package id mode.

File: conans/model/info.py [hunk 2]
```diff
@@ -513,26 +434,28 @@ def package_id(self):
         options and settings
         """
```
Comment (lasote) [inline conans/model/info.py:435]: just to make sure, this is going to change the package_id for the currently built packages (in the v2 repository), right?

**Review by lasote — APPROVED** [review]

File: conans/model/info.py (hunk 2 above)
Comment (memsharded) [reply conans/model/info.py:435]: Most likely not, as it didnt change any package_id in the codebase. This change only kicks in when building executables, defined with the ``package_type = "application"``. If I understood correctly, @czoido is faking the cmake package build at the moment, so shouldn't be an issue.

File: conans/client/graph/compute_pid.py (hunk 1 above)
Comment (memsharded) [reply conans/client/graph/compute_pid.py]: Renamed to ``dep_package_id_mode``

**Pull request merged**